### PR TITLE
fix(ingest/kafka-connect) support newer version of debezium

### PR DESCRIPTION
**Pull request by jaegwonseo**
## Summary
newer version of debezium kafka connect config is changed 
```
database.server.name -> topic.prefix
```
Separating the config used by checking the debezium connector version

document link
https://debezium.io/documentation/reference/1.9/connectors/mysql.html#mysql-property-database-server-name
https://debezium.io/documentation/reference/2.1/connectors/mysql.html#mysql-property-topic-prefix

## Issue
Closes https://github.com/datahub-project/datahub/issues/6927

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py [hunk 1]
```diff
@@ -640,59 +641,62 @@ def get_parser(
         connector_manifest: ConnectorManifest,
     ) -> DebeziumParser:
         connector_class = connector_manifest.config.get("connector.class", "")
+        # debezium version convention : x.y.z.{Final, Alpha, Beta, CR}
+        connector_version = int(self.version.split('.')[0])
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:645]: Is version always present ? Can we handle for missing version (i.e. version=None)

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py [hunk 2]
```diff
@@ -999,8 +1009,9 @@ def get_connectors_manifest(self) -> List[ConnectorManifest]:
                 elif connector_manifest.config.get("connector.class", "").startswith(
                     "io.debezium.connector"
                 ):
+                    connector_version = connector_version_dict.get(connector_manifest.config.get("connector.class", ""))
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:1012]:
```suggestion
                    connector_version = connector_version_dict.get(connector_manifest.config["connector.class"])
```

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py (hunk 1 above)
Comment (jaegwonseo) [reply metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:645]: newer version of debezium(2.1) is released
https://debezium.io/releases/

Comment (jaegwonseo) [reply metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:645]: here is rest api return values sample 
```json
[
  {
    "class": "com.wepay.kafka.connect.bigquery.BigQuerySinkConnector",
    "type": "sink",
    "version": "unknown"
  },
  {
    "class": "io.confluent.kafka.connect.datagen.DatagenConnector",
    "type": "source",
    "version": "null"
  },
  {
    "class": "io.debezium.connector.mysql.MySqlConnector",
    "type": "source",
    "version": "2.0.0.Final"
  }
]
```

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:645]: right. let's do this. If the call to conector-plugins api fails due to some reason(this is new but believe me we have seen this happen!) OR if api succeeds and version is none:
- fall back to old behavior .i.e older version (just for backward compatibility)

**Review by hsheth2 — CHANGES_REQUESTED** [review]
Can we do this using something like this instead, where we check topic.prefix and fallback to database.server.name if not available?

That way we don't even need to check the plugin version number

```py
# comment on why this was needed - similar to the PR description
server_name = connector_manifest.config.get(
    "topic.prefix", connector_manifest.config.get("database.server.name")
)
```

**Comment by jaegwonseo** [comment]
> Can we do this using something like this instead, where we check topic.prefix and fallback to database.server.name if not available?
> 
> That way we don't even need to check the plugin version number
> 
> ```python
> # comment on why this was needed - similar to the PR description
> server_name = connector_manifest.config.get(
>     "topic.prefix", connector_manifest.config.get("database.server.name")
> )
> ```

@hsheth2 
like you said
rather than calling the api to check the connector version,
it is better to use it after checking if the config exists.

@mayurinehate 
how about this

```py
    def get_server_name(self, connector_manifest: ConnectorManifest) -> str:
        if 'topic.prefix' in connector_manifest.config:
            return connector_manifest.config.get('topic.prefix')
        else:
            return connector_manifest.config.get('database.server.name')
```

**Review by hsheth2 — APPROVED** [review]

**Comment by jaegwonseo** [comment]
@hsheth2 
2 test failed 
cypress - frontend
trivy - spring boot jar security
Is this pr and test failure related?

**Review by mayurinehate — APPROVED** [review]

**Comment by mayurinehate** [comment]
I don't think the failures are related to changes in this PR, so we should be good to merge it.

**Pull request merged**